components/director/internal/domain/labeldef/service.go

```diff
@@ -60,3 +75,52 @@ func (s *service) List(ctx context.Context, tenant string) ([]model.LabelDefinit
 	}
 	return defs, nil
 }
+
+func (s *service) Update(ctx context.Context, def model.LabelDefinition) (model.LabelDefinition, error) {
+	if err := def.ValidateForUpdate(); err != nil {
+		return model.LabelDefinition{}, errors.Wrap(err, "while validating Label Definition")
+	}
+
+	ld, err := s.repo.GetByKey(ctx, def.Tenant, def.Key)
+	if err != nil {
+		return model.LabelDefinition{}, errors.Wrap(err, "while receiving Label Definition")
+	}
+
+	if ld.ID == "" {
+		return model.LabelDefinition{}, fmt.Errorf("id of received label definition is empty")
+	}
+
+	id := ld.ID
+	ld = &def
+	ld.ID = id
+
+	existingLabels, err := s.labelRepo.ListByKey(ctx, def.Tenant, def.Key)
+
+	validator, err := jsonschema.NewValidatorFromRawSchema(*def.Schema)
+	if err != nil {
+		return model.LabelDefinition{}, errors.Wrap(err, "while creating validator for new schema")
+	}
+
+	for _, label := range existingLabels {
+		ok, err := validator.ValidateRaw(label)
+		if err != nil {
+			return model.LabelDefinition{}, errors.Wrap(err, "while validating existing labels against new schema")
+		}
+
+		if ok == false {
+			return model.LabelDefinition{}, fmt.Errorf("label with key %s is not valid against new schema", label.Key)
+		}
+
+	}
+
+	if err := s.repo.Update(ctx, *ld); err != nil {
+		return model.LabelDefinition{}, errors.Wrap(err, "while updating Label Definition")
+	}
+
+	updatedLabelDef, err := s.repo.GetByKey(ctx, ld.Tenant, ld.Key)
```

Comment:
Is this call really necessary? Can't we just return the object that was just used in the repository Update call?

Reply:
removed that call